modules/service/src/main/scala/lucuma/odb/graphql/table/ObservationView.scala

```diff
@@ -53,6 +53,21 @@ trait ObservationView[F[_]] { self: SkunkMapping[F] =>
           }
         }
       }
+
+      object ScienceRequirements {
+        val Mode: ColumnRef = col("c_science_mode", science_mode.embedded)
+
+        object Spectroscopy {
+          val Wavelength         = col("c_spec_wavelength",          wavelength_pm.embedded)
+          val Resolution         = col("c_spec_resolution",          pos_int.opt)
```

Comment:
I'm not sure what `embedded` does or why it is necessary, to be honest.  @tpolecat ? I ended up switching `embedded` for `opt` when required by test case failures and it seems to depend upon whether the item is a simple field or one with field mappings for different units.

Reply:
`embedded` is a hack that's necessary when you have a column that represents a non-nullable field for a nullable embedded object. So it means "this column has to be null in the database schema but it's never null from the point of view of the GraphQL type it ends up in".